src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,98 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double INCREASING_COEFFICIENT = 1.5;
+    private int currentSize;
+    private T[] array;
+
+    public ArrayList() {
+        this.currentSize = 0;
+        this.array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        checkCapacity();
+        array[currentSize] = value;
+        currentSize++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkCapacity();
+        indexInRange(index, currentSize + 1);
+        System.arraycopy(array, index, array, index + 1, currentSize - index);
+        array[index] = value;
+        currentSize++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexInRange(index, currentSize);
+        return array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        indexInRange(index, currentSize);
+        array[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkCapacity();
+        indexInRange(index, currentSize);
+        final T removeElement = array[index];
+        System.arraycopy(array, index + 1, array, index, currentSize - index);
+        array[currentSize - 1] = null;
+        currentSize--;
+        return removeElement;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < currentSize; i++) {
+            if (element == array[i] || (element != null && element.equals(array[i]))) {
```

Comment:
The logical-AND operator ( && ) has higher precedence than the logical-OR operator ( || )
```suggestion
            if (element == array[i] || element != null && element.equals(array[i])) {
```